validator/client/attest.go

```diff
@@ -237,12 +243,24 @@ func (v *validator) waitToSlotOneThird(ctx context.Context, slot uint64) {
 	}
 	t := time.NewTimer(wait)
 	defer t.Stop()
-	select {
-	case <-ctx.Done():
-		traceutil.AnnotateError(span, ctx.Err())
-		return
-	case <-t.C:
-		return
+
+	bChannel := make(chan *ethpb.SignedBeaconBlock, 1)
+	sub := v.blockFeed.Subscribe(bChannel)
+	defer sub.Unsubscribe()
+
+	for {
```

Comment:
Question: is the for loop here really needed? doesn't it work to switch the order of the `defer` directive and only have the `select` statement?

Reply:
I think the for loop is not required. Nice catch

Reply:
Are you sure? What if `slot > b.Block.Slot` and timer hasn't expired

Reply:
Yep. All of the select cases return, so it would never loop more than once. 

On the first iteration, the goroutine will block until one of the select cases matches, then continue. In this case, it would return in every case so there would never be a second iteration of the for loop.

Reply:
We could have a second iteration if received block slot is less than input slot?
```go
select {
		case b := <-bChannel:
			if slot <= b.Block.Slot {
				return
			}
	}
```

Reply:
ah, yes. OK then we need it if that is the case. Thanks!